os/arch/arm/src/armv8-m/up_mpu.c

```diff
@@ -187,3 +187,35 @@ void up_mpu_set_register(uint32_t *mpu_regs)
 		putreg32(mpu_regs[REG_RASR], MPU_RLAR);
 	}
 }
+
+/****************************************************************************
+ * Name: up_mpu_disable_region
+ *
+ * Description:
+ *   Update tcb's mpu register values to disable a mpu region.
+ *   MPU region to be disabled is identified using tcb's mpu registers passed.
+ *   So, if the current MPU h/w settings for this MPU region match to those
+ *   of registers passed, set these new register values to mpu h/w as well.
+ *
+ ****************************************************************************/
+void up_mpu_disable_region(uint32_t *mpu_regs)
+{
+	uint32_t regval;
+
```

Comment:
If this api gets called for a kernel task, then no operation must be performed. We must return immediately. 
Please add below check for this. A null value in RBAR indicates a kernel task. 

`if (!mpu_regs[REG_RBAR]) {`
 `       return;`
`}`

Reply:
Done.